logic/src/commonMain/kotlin/com/wire/kalium/logic/feature/message/EphemeralNotificationsManager.kt

```diff
@@ -0,0 +1,27 @@
+package com.wire.kalium.logic.feature.message
+
+import com.wire.kalium.logic.data.notification.LocalNotificationConversation
+import kotlinx.coroutines.channels.Channel
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.consumeAsFlow
+import kotlinx.coroutines.flow.emptyFlow
+
+/**
+ * This singleton allow us to queue ephemeral notifications from different user flows.
+ * Ideally we should have logic that allows to mark messages as notified, but this will act for cases when we need to notify the user on
+ * information we don't have available persisted.
+ */
+object EphemeralNotificationsManager {
+
+    private val notifications =
+        Channel<LocalNotificationConversation>(capacity = Channel.CONFLATED) { emptyFlow<LocalNotificationConversation>() }
+
+    fun observeEphemeralNotifications(): Flow<LocalNotificationConversation> {
+        return notifications.consumeAsFlow()
+    }
+
+    suspend fun scheduleNotification(localNotificationConversation: LocalNotificationConversation) {
```

Comment:
This seems to push the notification directly instead of scheduling right? or are we planning to include some sort of delay later on?

Reply:
This will just emit a new event to the `Channel` and after that the `NotificationManager` (on Reloaded) will handle the displaying of it, so there can be a delay, but to my knowledge will depend on the notification manager and the system itself when to display it.